FEXCore/Source/Interface/Core/Core.cpp

```diff
@@ -908,6 +907,10 @@ uintptr_t ContextImpl::CompileSingleStep(FEXCore::Core::CpuStateFrame* Frame, ui
 }
 
 static void InvalidateGuestThreadCodeRange(FEXCore::Core::InternalThreadState* Thread, uint64_t Start, uint64_t Length) {
+  // Safe as the thread's code invalidation mutex must be locked here, ensures now-modified mappings aren't cached as
+  // being in their previous state.
```

Comment:
This is difficult to understand due to using fragments instead of full sentences. Also you're explaining why this is safe but it's unclear what the concern is to begin with.